cmd/lightclient/sentinel/request.go

```diff
@@ -0,0 +1,89 @@
+package sentinel
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/ledgerwatch/erigon/cmd/lightclient/sentinel/handlers"
+	"github.com/ledgerwatch/erigon/cmd/lightclient/sentinel/proto"
+	"github.com/ledgerwatch/erigon/cmd/lightclient/sentinel/proto/p2p"
+	"github.com/ledgerwatch/erigon/cmd/lightclient/sentinel/proto/ssz_snappy"
+	"github.com/ledgerwatch/log/v3"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+)
+
+func (s *Sentinel) sendPingReqV1() (proto.Packet, error) {
+	requestPacket := &p2p.Ping{
+		Id: uint64(1),
+	}
+	responsePacket := &p2p.Ping{}
+	return sendRequest(s, requestPacket, responsePacket, handlers.PingProtocolV1)
+}
+
+func (s *Sentinel) sendMetadataReqV1() (proto.Packet, error) {
+	requestPacket := &p2p.MetadataV1{}
+	responsePacket := &p2p.MetadataV1{}
+
+	return sendRequest(s, requestPacket, responsePacket, handlers.MedataProtocolV1)
+}
+
+// TODO: add the request of the request topics
+
+func sendRequest(s *Sentinel, requestPacket proto.Packet, responsePacket proto.Packet, topic string) (proto.Packet, error) {
+	_, peerInfo, err := connectToRandomPeer(s)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to a random peer err=%s", err)
+	}
+
+	peerId := peerInfo.ID
+	stream, err := s.host.NewStream(s.ctx, peerId, protocol.ID(topic))
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to begin stream, err=%s", err)
+	}
+	sc := ssz_snappy.NewStreamCodec(stream)
+	defer sc.Close()
+
+	if _, err := sc.WritePacket(requestPacket); err != nil {
+		return nil, fmt.Errorf("failed to write packet type=%s, err=%s", reflect.TypeOf(requestPacket), err)
+	}
+
+	if err := sc.CloseWriter(); err != nil {
+		return nil, fmt.Errorf("failed to close write stream, err=%s", err)
+	}
+	log.Info("[Req] sent request", "topic", topic, "peer", peerId)
+
+	time.Sleep(1 * time.Second)
```

Comment:
try with something more flexible, try making a for loop and try again for 15 seconds with intervals of 10 Milliseconds